File: January Challenge/Java/Daily Temperatures.java
import java.util.Stack;
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int [] res = new int[n];
        Stack<Integer> st = new Stack();

        for(int i = 0; i < n; i++){
            while (!st.empty() && temperatures[st.peek()] < temperatures[i]){
                int ind = st.pop();
                res[ind] = i - ind;
            }
            st.push(i);
        }
        return res;
    }
}

File: January Challenge/Python/Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        st = []

        for i, temps in enumerate(temperatures):
            while st and temperatures[st[-1]] < temps:
                ind = st.pop()
                res[ind] = i - ind
            st.append(i)
        return res
        
